spring-integration-test-support/src/main/java/org/springframework/integration/test/matcher/HeaderMatcher.java

```diff
@@ -68,103 +68,99 @@
  *
  * @author Alex Peters
  * @author Iwein Fuld
+ * @author Artem Bilan
  *
  */
-public class HeaderMatcher extends TypeSafeMatcher<Message<?>> {
+@SuppressWarnings("rawtypes")
+public class HeaderMatcher extends TypeSafeMatcher<Message> {
 
 	private final Matcher<?> matcher;
 
 	/**
-	 * @param matcher
+	 * @param matcher the target matcher to delegate
 	 */
-	HeaderMatcher(Matcher<?> matcher) {
+	private HeaderMatcher(Matcher<?> matcher) {
 		super();
 		this.matcher = matcher;
 	}
 
-	/**
-	 * {@inheritDoc}
-	 */
 	@Override
-	public boolean matchesSafely(Message<?> item) {
+	public boolean matchesSafely(Message item) {
```

Comment:
Otherwise it doesn't compile with the `MockMessageHandler.assertNext()`.
I tried to fix `PayloadMatcher` to be generic-aware in the one of previous commits, but that fell to the issue with undesired casts in the target test code when we would like just deal with `Matcher`s.

Since I would like to make `MockMessageHandler.assertNext()` compatible with both `HeaderMatcher` and `PayloadMatcher`, plus `PayloadAndHeaderMatcher`, there is no way unless make everything generic-aware or raw types.

Any other ideas?

Thanks for pointing to this anyway!